File: src/main/java/services/PackageService.java
package services;

import dao.PackageDao;
import models.File;
import models.Package;

import java.util.List;

public class PackageService extends Service<Package> {

    private PackageDao packageDao;

    public PackageService(int userID, String password) {
        packageDao = new PackageDao(userID, password);
    }

    public Package find(int id) {
        return packageDao.findById(id);
    }

    public void save(Package _package) {
        packageDao.save(_package);
    }

    public void update(Package _package) {
        packageDao.update(_package);
    }

    public void delete(Package _package) {
        packageDao.delete(_package);
    }

    public File findFileById(int id) {
        return packageDao.findFileById(id);
    }

    public List<Package> getAll() {
        return packageDao.getAll();
    }
}

File: src/main/kotlin/services/PackageService.kt
package services

import dao.PackageDao
import models.File
import models.Package

class PackageService(userID: Int, password: String) : Service<Package>(userID, password) {

    private var packageDao: PackageDao = PackageDao(userID, password)

    override fun reload(): PackageService {
        packageDao = PackageDao(userID, password)
        return this
    }

    override fun find(id: Int): Package? {
        return packageDao.findById(id)
    }

    override fun save(obj: Package) {
        packageDao.save(obj)
    }

    override fun update(obj: Package) {
        packageDao.update(obj)
    }

    override fun delete(obj: Package) {
        packageDao.delete(obj)
    }

    fun findFileById(id: Int): File {
        return packageDao.findFileById(id)
    }

    override fun getAll(): List<Package>? {
        return packageDao.all?.drop(1)
    }
}
